src/persist-types/src/schema.rs

```diff
@@ -199,7 +213,14 @@ fn backward_compatible_typ(old: &DataType, new: &DataType) -> Option<ArrayMigrat
         (FixedSizeBinary(_), _) => None,
         (Struct(o), Struct(n)) => backward_compatible_struct(o, n),
         (Struct(_), _) => None,
-        (List(o), List(n)) => (o == n).then_some(NoOp),
+        (List(o), List(n)) | (Map(o, _), List(n)) => {
+            if o.is_nullable() && !n.is_nullable() {
```

Comment:
Does this cover the MakeNullable case? If so, how (probably deserves a comment with the answer)?